packages/twenty-server/src/engine/core-modules/imap-connection/dtos/imap-connection.dto.ts

```diff
@@ -0,0 +1,84 @@
+import { Field, InputType } from '@nestjs/graphql';
+
+import {
+  IsEnum,
+  IsNotEmpty,
+  IsNumber,
+  IsOptional,
+  IsString,
+  IsUUID,
+  Min,
+} from 'class-validator';
+
+import { MessageChannelVisibility } from 'src/modules/messaging/common/standard-objects/message-channel.workspace-entity';
+
+@InputType()
+export class TestImapConnectionInput {
+  @Field(() => String)
+  @IsNotEmpty()
+  @IsString()
+  host: string;
+
+  @Field(() => Number)
+  @IsNotEmpty()
+  @IsNumber()
+  @Min(1)
+  port: number;
+
+  @Field(() => Boolean)
+  @IsNotEmpty()
+  secure: boolean;
+
+  @Field(() => String)
+  @IsNotEmpty()
+  @IsString()
+  handle: string;
+
+  @Field(() => String)
+  @IsNotEmpty()
+  @IsString()
+  password: string;
```

Comment:
FYI we sometimes encrypt in DB these kind of things. See for example encryptPassword in RemoteServerService. I'm OK not to do it, I don't think it adds that much protection - if someone can access the DB we're probably not in a good position no matter what 😅

Reply:
Yes indeed, that's what I meant when I said in any case if someone access the DB we're probably already in a very bad position so I don't think it's worth it. But it's worth adding it as a comment on the review for future reader to show it was explicitly debated :)